language/Records.md

```diff
@@ -79,6 +88,65 @@ A record update function can also be defined by using an `_` inplace of the reco
 _ { fieldName = newValue }
 ```
 
+## Record Puns
+
+_Record puns_ enable concise code when record fields have the same name as other values. This feature is useful for constructing records and pattern matching.
+
+### Constructing
+
+```purs
+origin :: Point
+origin = { x, y }
+-- origin = { x: x, y: y } -- Equivalent
+  where
+    x = 0.0
+    y = 0.0
+```
+
+### Pattern Matching
+
+```purs
+showPoint :: Point -> String
+showPoint { x, y } = show x <> ", " <> show y
+-- showPoint { x : x, y : y } = show x <> ", " <> show y -- Equivalent
+```
+
+### Not for Record Updates
+
+Note that puns may not be used for record updates.
+
+```purescript
+setX :: Number -> Point -> Point
+setX x point = point { x = x }
+-- setX x point = point { x } -- Not allowed
+```
+
+## Merging Records
+
+The [`record`](https://pursuit.purescript.org/packages/purescript-record) package enables additional record operations, such as `merge` and `union`.
+
+Here's an example of using the `disjointUnion` function to add a `z` field to a `Point` record. The `to3d` function merges the original record `p` with another record `{ z }` created from this new `Number`:
+
+```purescript
+import Record (disjointUnion)
+
+type Point3d
+  = { x :: Number
+    , y :: Number
+    , z :: Number
+    }
```

Comment:
I think typical syntax in this repository places the `=` on the same line as the `type` keyword:

```purs
type Point3d =
  { x :: Number
  , y :: Number
  , z :: Number
  }
```

Reply:
Applied suggestion